---
title: "Pre-oligotyping diagnostics for functional genes"
author: "Marina Zure and Antonio Fernandez-Guerra"
date: "February 29, 2016"
output:
  html_document:
    toc: yes
---

**How to decide if we should keep the third codon position if we want to perform a oligotyping analysis on functional genes**

In this examples the files are used to perform a Minimum Entropy Decomposition ([MED](http://merenlab.org/projects/med/)) analysis, the unsupervised flavour for [oligotyping](http://merenlab.org/projects/oligotyping/).

First we load the package:
```{r}
library(oligo4fun)
```

Let's have a look to the loaded data, first to the entropy file:
```{r}
entropy
```
and to the MED input alignment file:
```{r}
aln
```
One can read any fasta file using read.dna function from [ape](https://cran.r-project.org/web/packages/ape/index.html):

```{r, eval=FALSE}
aln <- read.dna(file=file, format = "fasta", as.character=FALSE, skip=0)
```

First we will explore the entropy value distribution on all codon positions:
```{r}
entropy_plot <- plot_entropy_density(entropy)
entropy_plot
```

The resulting plot shows that the 3rd codon position has higher entropy values suggesting
that if we use this aligment as input for the MED it could result in a possible overestimation
of the diversity. 

Next step is to know if the higher entropy values implies a real change in the aminoacid sequence
or are silent substitution. We will use a saturation plot to explore the transitions and transversions ratio. As the NGS datasets that can be used for MED are really large, in this example we will pick 500 random sequences for the analysis (we recommend to use 5000 sequences aprox). For the distance calculation we use the _Kimura's 2-parameters distance_ (K80). We set all = FALSE because we only want to focus on the 3rd codon position, we can calculate the ratio for all codon positions with ALL = TRUE.

```{r}
saturation_plot <- plot_saturation(aln, nseqs = 500, model = "K80", all = FALSE )
```

We can visually inspect the plot:
```{r}
saturation_plot$plot
```

Get some statistics:
```{r}
saturation_plot$stats
```

Or get the alignment without the 3rd codon position that can be used for the MED analysis:
```{r}
saturation_plot$aln_no_3rd
```

But using only 500 or 5000 sequences is not enough to be sure that there is no substitution saturation in our alignment. In this case one can use the function **plot_saturation_n** to perform n random subsamplings to the main alignment file. An example using the sequential version with performing 10 random subsamplings of 500 sequences:
```{r, echo = FALSE}
random_saturation_plots <- plot_saturation_n(aln, rep = 10, nseqs = 500, model = "K80", parallel = FALSE, all = FALSE)
```

For a real world analysis we would perform at least 100 random subsamplings of >5000 sequences. The parallel version uses [https://github.com/tudo-r/BatchJobs](BatchJobs) to distribute the different random subsamplings over multiple cores or a computer cluster. An example using the different cores of a computer:

```{r}
random_saturation_plots <- plot_saturation_n(aln, rep = 10, nseqs = 500, model = "K80", parallel = TRUE, all = FALSE, conf_file = "~/.BatchJobs.R", reg_id = "rhodo", reg_dir = "~/rhodo_dir")
```

> For reproducibility purposes, all seeds used by **plot_saturation** and **plot_saturation_n** are stored in the object returned under **seed**.

The contents of the .BatchJobs.R file:
```{r, eval = FALSE}
cluster.functions = makeClusterFunctionsMulticore(ncpus = max(getOption("mc.cores", parallel::detectCores()) - 1L, 1L))
staged.queries = TRUE
debug = FALSE
```

You can find examples for other queue managers [here](https://github.com/tudo-r/BatchJobs/tree/master/examples/). For example, if our cluster uses SGE, our **.tmpl** file should be similar to this [here](https://github.com/tudo-r/BatchJobs/blob/master/examples/cfSGE/simple.tmpl) and the .BatchJobs.R file should contain:
```{r, eval = FALSE}
cluster.functions = makeClusterFunctionsSGE("simple.tmpl")
staged.queries = TRUE
debug = FALSE
```

Once we got the results we can explore the object returned. We can explore the combined statistics:
```{r}
random_saturation_plots$combined_stats
```

Or we can look at the distribution of the average values for transitions and transversion for all random subsamplings:
```{r}
write_saturation_plot_density(random_saturation_plots, dir = "~", format = "pdf", show = TRUE)
```

This plot also suggests that there are more transitions than transversions so would be recommendable to remove the 3rd codon position from the alignment.

We can easily do this:
```{r}
write_saturation_alignments(random_saturation_plots, dir = "~", no3rd = TRUE)
```

We can save all plots in disk:
```{r}
write_saturation_plots(random_saturation_plots, dir = "~", format = "pdf")
```

And all random alignments used:
```{r}
write_saturation_alignments(random_saturation_plots, dir = "~")
```

After all this steps we can use the exported alignment for the [MED](http://merenlab.org/projects/med/) analysis.
