<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="description" content="oligo4fun : Pre-oligotyping diagnostics for functional genes">

    <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">

    <title>oligo4fun</title>
  </head>

  <body>

    <!-- HEADER -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <a id="forkme_banner" href="https://github.com/genomewalker/oligo4fun">View on GitHub</a>

          <h1 id="project_title">oligo4fun</h1>
          <h2 id="project_tagline">Pre-oligotyping diagnostics for functional genes</h2>

            <section id="downloads">
              <a class="zip_download_link" href="https://github.com/genomewalker/oligo4fun/zipball/master">Download this project as a .zip file</a>
              <a class="tar_download_link" href="https://github.com/genomewalker/oligo4fun/tarball/master">Download this project as a tar.gz file</a>
            </section>
        </header>
    </div>

    <!-- MAIN CONTENT -->
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner">
        <h3>
<a id="background" class="anchor" href="#background" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Background</h3>

<p><a href="http://merenlab.org/projects/oligotyping/"><strong>Oligotyping</strong></a> [1] was originally developed to detect subtle differences in the 16S rRNA gene, which is a non-protein coding gene, conserved on the nucleotide level and under functional selection on all nucleotide positions. On the contrary, protein coding genes are conserved on the amino acid level and changes at the nucleotide level are often non-selective in the third codon due to the degenerated nature of the genetic code (silent substitutions). Changes on the first or second codon are likely to produce a change on the amino acidic sequence, and in some cases this change could result in an increase of fitness or adaptation. Using a high resolution entropy based method like <a href="http://merenlab.org/projects/oligotyping/"><strong>oligotyping</strong></a> or <a href="http://merenlab.org/projects/med/"><strong>MED</strong></a> [2] on coding sequences can result in the overestimation of the effective genetic diversity in the environment as a result of a much faster evolution rate of the third position in the codon. </p>

<p>In Zure et al. 2016 [3] we developed a set of diagnostics to explore the existence of substitution saturation on the alignments of functional genes that will be used for oligotyping.</p>

<h3>
<a id="tutorial" class="anchor" href="#tutorial" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Tutorial</h3>

<p>A detailed tutorial on how to use <strong>oligo4fun</strong> can be found <a href="http://genomewalker.github.io/oligo4fun/tutorial/oligo4fun.html"><strong>here</strong></a></p>

<h3>
<a id="installation" class="anchor" href="#installation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Installation</h3>

<p>Detailed instructions for the installation can be found <a href="http://github.com/genomewalker/oligo4fun/"><strong>here</strong></a></p>


<h3>
<a id="references" class="anchor" href="#references" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>References</h3>

<p>[1] Eren, A. M., Maignien, L., Sul, W. J., Murphy, L. G., Grim, S. L., Morrison, H. G., and Sogin, M. L. (2013). Oligotyping: Differentiating between closely related microbial taxa using 16S rRNA gene data. Methods Ecol Evol. 4(12)</p>

<p>[2] Eren, A. M., Morrison, H. G., Lescault, P. J., Reveillaud, J., Vineis, J. H., &amp; Sogin, M. L. (2015). Minimum entropy decomposition: Unsupervised oligotyping for sensitive partitioning of high-throughput marker gene sequences. The ISME Journal, 9(4), 968-979</p>

<p>[3] Zure, M., Fernandez-Guerra, A., Munn C. and Harder, J. (2016). Geographic distribution at a sub-species resolution level: closely related <i>Rhodopirellula</i> species in European coastal sediment. Accepted in ISMEJ.</p>
      </section>
    </div>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <p>Published with <a href="https://pages.github.com">GitHub Pages</a></p>
      </footer>
    </div>

    

  </body>
</html>
